File: agents/docs/docsAgent.ts
import { Agent } from '../base';

/** A dummy documentation agent */
export class DocsAgent implements Agent {
  readonly name = 'docs';

  async run(task: string): Promise<string> {
    console.log(`[DocsAgent] processing: ${task}`);
    try {
      await delay(100);
      return `Docs response for: ${task}`;
    } catch (err) {
      console.error('DocsAgent error:', err);
      throw err;
    }
  }

  getCapabilities(): string[] {
    return ['documentation'];
  }
}

async function delay(ms: number) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

File: codeorbit/agents/docs/docs_agent.rs
use crate::core::agent_registry::Agent;
use crate::core::error::Result;
use async_trait::async_trait;

/// A dummy documentation agent simulating doc generation
pub struct DocsAgent {
    id: String,
}

impl DocsAgent {
    /// Create a new `DocsAgent` instance
    pub fn new() -> Self {
        Self {
            id: "docs".to_string(),
        }
    }
}

#[async_trait]
impl Agent for DocsAgent {
    fn id(&self) -> &str {
        &self.id
    }

    async fn initialize(&mut self) -> Result<()> {
        log::info!("Initializing DocsAgent");
        Ok(())
    }

    async fn process(&self, request: &str) -> Result<String> {
        log::debug!("DocsAgent processing: {}", request);
        Ok(format!("Docs response for: {}", request))
    }

    async fn shutdown(&self) -> Result<()> {
        Ok(())
    }
}
